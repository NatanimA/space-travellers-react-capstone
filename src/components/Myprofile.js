import React from 'react';
import ListGroup from 'react-bootstrap/ListGroup';
import Alert from 'react-bootstrap/Alert';
import { useSelector } from 'react-redux';

const Myprofile = () => {
  const rockets = useSelector((state) => state.rockets).filter((val) => val.reserved === true);
  const missions = useSelector((state) => state.missions)
    .filter((mission) => mission.member === true);

  return (
    <section className="section-containers section-rockets profile__container">
      <div>
        <h1>My Missions</h1>
        <ListGroup>
          {
            (missions.length > 0)
              ? missions.map((mission) => (
                <ListGroup.Item key={mission.mission_id}>{mission.mission_name}</ListGroup.Item>
              )) : (
                <Alert variant="danger">
                  <Alert.Heading>Oh sorry! You&apos;ve not joined any missions!</Alert.Heading>
                </Alert>
              )
          }
        </ListGroup>
      </div>
      <div>
        <h1>My Rockets</h1>
        <ListGroup>
          {
            (rockets.length > 0)
              ? rockets.map((rockets) => (
                <ListGroup.Item key={rockets.id}>{rockets.name}</ListGroup.Item>
              )) : (
                <Alert variant="danger">
                  <Alert.Heading>Oh sorry! You&apos;ve not reserved any rocket!</Alert.Heading>
                </Alert>
              )
          }
        </ListGroup>
      </div>
    </section>
  );
};

export default Myprofile;
