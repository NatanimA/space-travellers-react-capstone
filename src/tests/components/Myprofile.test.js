import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Myprofile from '../../components/Myprofile';

test('renders rockets component snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Myprofile />
      </BrowserRouter>
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
