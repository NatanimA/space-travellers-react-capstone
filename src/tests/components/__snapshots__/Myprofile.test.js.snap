// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders rockets component snapshot 1`] = `
<DocumentFragment>
  <section
    class="section-containers section-rockets profile__container"
  >
    <div>
      <h1>
        My Missions
      </h1>
      <div
        class="list-group"
      >
        <div
          class="fade alert alert-danger show"
          role="alert"
        >
          <div
            class="alert-heading h4"
          >
            Oh sorry! You've not joined any missions!
          </div>
        </div>
      </div>
    </div>
    <div>
      <h1>
        My Rockets
      </h1>
      <div
        class="list-group"
      >
        <div
          class="fade alert alert-danger show"
          role="alert"
        >
          <div
            class="alert-heading h4"
          >
            Oh sorry! You've not reserved any rocket!
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
